AutoCollection/HttpRequestParser.ts

```diff
@@ -204,23 +247,27 @@ class HttpRequestParser extends RequestParser {
         if (request.headers) {
             this.correlationContextHeader = request.headers[RequestResponseHeaders.correlationContextHeader];
 
-            if (request.headers[RequestResponseHeaders.requestIdHeader]) {
-                this.parentId = request.headers[RequestResponseHeaders.requestIdHeader];
-                this.requestId = CorrelationIdManager.generateRequestId(this.parentId);
-                this.correlationContextHeader = request.headers[RequestResponseHeaders.correlationContextHeader];
+            if (CorrelationIdManager.w3cEnabled) {
+                this.parseW3CTraceHeaders(request, requestId);
             } else {
-                // Legacy fallback
-                const rootId = request.headers[RequestResponseHeaders.rootIdHeader];
-                this.parentId = request.headers[RequestResponseHeaders.parentIdHeader];
-                this.requestId = CorrelationIdManager.generateRequestId(rootId || this.parentId);
-                this.correlationContextHeader = null;
-            }
-            if (requestId) {
-                // For the scenarios that don't guarantee an AI-created context,
-                // override the requestId with the provided one.
-                this.requestId = requestId;
+                if (request.headers[RequestResponseHeaders.requestIdHeader]) {
+                    this.parentId = request.headers[RequestResponseHeaders.requestIdHeader];
```

Comment:
It's a bit strange to have this logic duplicated elsewhere in w3c mode - seems like a good way to have hard to find bugs. Any way we can unify some of this header parsing so headers of one kind only have one chunk of code that handles them?